### ISSUE TITLE
Install default NetworkPolicies for webhooks and gRPC communication

### ISSUE THREAD

**Issue by sttts** [crossplane/crossplane#4929]
### What happened?

When playing with conversion webhooks in providers, I noticed that there is no `NetworkPolicy` installed. I.e. depending on the network setup (i.e. which CNI), the apiserver might not be able to talk to the conversion webhook.

Probably the same is true for the admission webhook of Crossplane itself. That's more critical because this webhook is enabled in 1.14 by default (no provider is known today that deploys webhooks).

Compare https://kubernetes.io/docs/concepts/services-networking/network-policies/#prerequisites for the concept.

~~This has 1.14 release blocker potential and we should check the impact.~~

Update:

We clarified in Slack: the admission webhook is installed since 1.11, just not in enforcing mode. So the problem is not new. We should definitely document this or add automation to Crossplane. **But this is not critical for 1.14**.

### What environment did it happen in?
Crossplane version: 1.14

**Comment by phisco** [comment]
I wouldn't call it a bug, a user with network policies defined will be able to define and deploy them alongside Crossplane or a Provider. Even more so, now that we have merged https://github.com/crossplane/crossplane/pull/4664.

So, although this can already be done for Crossplane, I found a few examples of charts deploying NetworkPolicies if explicitly requested:
- https://github.com/prometheus-community/helm-charts/blob/84ab6297ac99330cf3dc3d9c68d811172582acaa/charts/kube-prometheus-stack/values.yaml#L2487
- https://github.com/cert-manager/cert-manager/blob/a8813c5f43730e312ae5ba1601b3dadd3f23e835/deploy/charts/cert-manager/values.yaml#L474C1-L475C1
We could think of adding this too, definitely not a bug though.

Doing so for Runtime Packages, such as Providers and Functions, would definitely require more work and given that we have a not so complex network graph for what concerns Crossplane resource communication as we enforce ports and usually deploy them all in a single namespace, we could probably just document it and leave to users to define NetworkPolicies on their own with the granularity they need.
